Description: Add "RequestShutdown" and "RequestReboot" DBus methods to allow
other applications to shutdown or reboot the machine via the session manager
Author: Chris Coulson <chrisccoulson@ubuntu.com>

Index: gnome-session-3.7.3/gnome-session/gsm-manager.c
===================================================================
--- gnome-session-3.7.3.orig/gnome-session/gsm-manager.c	2013-01-07 12:51:02.026999067 +1300
+++ gnome-session-3.7.3/gnome-session/gsm-manager.c	2013-01-07 12:51:02.022999066 +1300
@@ -3664,6 +3664,48 @@
 }
 
 gboolean
+gsm_manager_request_shutdown (GsmManager *manager,
+                              GError    **error)
+{
+        g_debug ("GsmManager: RequestShutdown called");
+
+        g_return_val_if_fail (GSM_IS_MANAGER (manager), FALSE);
+
+        if (manager->priv->phase != GSM_MANAGER_PHASE_RUNNING) {
+                g_set_error (error,
+                             GSM_MANAGER_ERROR,
+                             GSM_MANAGER_ERROR_NOT_IN_RUNNING,
+                             "RequestShutdown interface is only available during the Running phase");
+                return FALSE;
+        }
+
+        request_shutdown (manager);
+
+        return TRUE;
+}
+
+gboolean
+gsm_manager_request_reboot (GsmManager *manager,
+                            GError    **error)
+{
+        g_debug ("GsmManager: RequestReboot called");
+
+        g_return_val_if_fail (GSM_IS_MANAGER (manager), FALSE);
+
+        if (manager->priv->phase != GSM_MANAGER_PHASE_RUNNING) {
+                g_set_error (error,
+                             GSM_MANAGER_ERROR,
+                             GSM_MANAGER_ERROR_NOT_IN_RUNNING,
+                             "RequestReboot interface is only available during the Running phase");
+                return FALSE;
+        }
+
+        request_reboot (manager);
+
+        return TRUE;
+}
+
+gboolean
 gsm_manager_shutdown (GsmManager *manager,
                       GError    **error)
 {
Index: gnome-session-3.7.3/gnome-session/gsm-manager.h
===================================================================
--- gnome-session-3.7.3.orig/gnome-session/gsm-manager.h	2013-01-07 12:51:02.026999067 +1300
+++ gnome-session-3.7.3/gnome-session/gsm-manager.h	2013-01-07 12:51:02.022999066 +1300
@@ -160,7 +160,10 @@
                                                                 guint                  flags,
                                                                 gboolean              *is_inhibited,
                                                                 GError                *error);
-
+gboolean            gsm_manager_request_shutdown               (GsmManager     *manager,
+                                                                GError        **error);
+gboolean            gsm_manager_request_reboot                 (GsmManager     *manager,
+                                                                GError        **error);  
 gboolean            gsm_manager_shutdown                       (GsmManager     *manager,
                                                                 GError        **error);
 gboolean            gsm_manager_reboot                         (GsmManager     *manager,
Index: gnome-session-3.7.3/gnome-session/org.gnome.SessionManager.xml
===================================================================
--- gnome-session-3.7.3.orig/gnome-session/org.gnome.SessionManager.xml	2013-01-07 12:51:02.026999067 +1300
+++ gnome-session-3.7.3/gnome-session/org.gnome.SessionManager.xml	2013-01-07 12:51:02.022999066 +1300
@@ -346,6 +346,23 @@
       </doc:doc>
     </method>
 
+	<method name="RequestShutdown">
+      <doc:doc>
+        <doc:description>
+          <doc:para>Request a shutdown with no dialog</doc:para>
+        </doc:description>
+      </doc:doc>
+	</method>
+
+	<method name="RequestReboot">
+      <doc:doc>
+        <doc:description>
+          <doc:para>Request a reboot with no dialog</doc:para>
+        </doc:description>
+      </doc:doc>
+	</method>
+	
+
     <!-- Signals -->
 
     <signal name="ClientAdded">
